src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -152,6 +154,88 @@ private MethodBase FunctionLookup(IList args)
             return mi;
         }
 
+        private static HashSet<Type> GetTypeStatisticsForArray(ExprListNode array)
+        {
+            var arrayTypes = new HashSet<Type>();
+
+            foreach (var exp in array.Exprs)
+            {
+                if (exp is ExprListNode eln)
+                {
+                    var subArray = GetTypeStatisticsForArray(eln);
+                    var t = GetOverallTypeForArray(subArray);
+
+                    if (t != typeof(object))
+                    {
+                        t = t.MakeArrayType();
+                    }
+                    arrayTypes.Add(t);
```

Comment:
why not make the array type in the case of object too ?
type would be : `object[]`

Reply:
I'm not sure why I added this condition. Looks like I can safely make it `object[]` as well. Will try that.